lib/ueberauth/strategies/helpers.ex

```diff
@@ -125,13 +125,19 @@ defmodule Ueberauth.Strategy.Helpers do
     %URI{
       host: conn.host,
       scheme: to_string(conn.scheme),
-      port: conn.port,
+      port: normalize_port(conn.scheme, conn.port),
       path: path,
-      query: URI.encode_query(opts)
+      query: encode_query(opts)
     }
     |> to_string
   end
 
+  defp normalize_port(:https, 80), do: 443
```

Comment:
I don't think this will every be a string as `Plug` and the `URI` module use an atom.